File: app/src/main/java/linusfessler/alarmtiles/dao/AlarmDao.java
package linusfessler.alarmtiles.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import linusfessler.alarmtiles.model.AlarmTile;

@Dao
public interface AlarmDao {

    @Query("SELECT * FROM AlarmTile")
    LiveData<AlarmTile> select();

    @Insert
    void insert(AlarmTile alarmTile);

    @Update
    void update(AlarmTile alarmTile);

    @Query("SELECT COUNT(*) FROM AlarmTile")
    int count();
}

File: app/src/main/java/linusfessler/alarmtiles/shared/alarm/AlarmDao.kt
package linusfessler.alarmtiles.shared.alarm

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import io.reactivex.Single

@Dao
interface AlarmDao {
    @Query("SELECT * FROM Alarm WHERE id = :id")
    fun select(id: Long): Single<Alarm>

    @Insert
    fun insert(alarm: Alarm)

    @Update
    fun update(alarm: Alarm)
}